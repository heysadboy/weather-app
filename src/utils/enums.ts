export enum EActionType {
    fetch_weather = 'FETCH_WEATHER',
    temp_data = 'SET_TEMP_DATA',
    left_page = 'LEFT_PAGE',
    right_page = 'RIGHT_PAGE'
}

export enum EStatusType {
    error = 'ERROR',
    loading = 'LOADING',
    ok = 'OK'
}

export enum ETempType {
    f = "Fahrenheit",
    c = "Celsius"
}