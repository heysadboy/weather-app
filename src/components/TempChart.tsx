import { FC } from "react";
import { connect } from "react-redux";
import { BarChart, Bar, XAxis } from "recharts";
import { bindActionCreators } from "redux";
import { ThunkDispatch } from "redux-thunk";
import { tempData } from "../actions";
import '../css/TempChart.css'
import { ETempType } from "../utils/enums";
import { IAction, IWeather } from "../utils/interfaces";
import { AppState } from "../utils/types";

interface ITempChartProp {
    weather: IWeather[] | [],
    tempType: ETempType,
    currentDay: string
}

const mapStateToProps = (state: AppState) => ({
    weather: state.weather,
});

const mapDispatchToProps = (
    dispatch: ThunkDispatch<AppState, {}, IAction>
) => ({
    tempData: bindActionCreators(tempData, dispatch)
});

const TempChart: FC<ITempChartProp> = ({ tempType, weather, currentDay }) => {
    const getTempKey = () => {
        if (tempType === ETempType.f) {
            return "temp_f";
        }
        else {
            return "temp_c";
        }
    }

    const barData = weather.filter((weather_item) => {
        return (weather_item.dt_txt === currentDay)
    });

    return (
        <div id="temp-container" className="ui center aligned grid">
            <BarChart id="temp-chart" width={800} height={500} data={barData}>
                <XAxis dataKey={getTempKey()} />
                <Bar dataKey={getTempKey()} fill="#424242" />
            </BarChart>

        </div>
    );
}

export default connect(mapStateToProps, mapDispatchToProps)(TempChart);