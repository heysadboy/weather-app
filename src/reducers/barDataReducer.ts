import { EActionType } from '../utils/enums';
import { IAction } from '../utils/interfaces';


const barDataReducer = (state = [], action: IAction) => {
    switch (action.type) {
        case EActionType.barchart_data:
            return action.payload;
        default:
            return state;
    }
};

export default barDataReducer;