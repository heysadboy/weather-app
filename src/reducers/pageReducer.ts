import { EActionType } from '../utils/enums';
import { IAction } from '../utils/interfaces';
import { INITIAL_PAGE } from '../utils/data';

const pageReducer = (state = INITIAL_PAGE, action: IAction) => {
    switch (action.type) {
        case EActionType.left_page:
            if (state.start > 0) {
                return { ...state, start: state.start - 1, end: state.end - 1 };
            }
            else {
                return state;
            }

        case EActionType.right_page:
            if (state.end < 5) {
                return { ...state, start: state.start + 1, end: state.end + 1 };
            }
            else {
                return state;
            }
        default:
            return state;
    }
}

export default pageReducer;