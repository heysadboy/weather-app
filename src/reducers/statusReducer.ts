import { INITAIL_STATUS } from '../utils/data';
import { EStatusType } from '../utils/enums';
import { IAction } from '../utils/interfaces';

const statusReducer = (state = {INITAIL_STATUS}, action: IAction) => {
    switch (action.type) {
        case EStatusType.error:
            return action.payload;
        case EStatusType.ok:
            return action.payload;
        case EStatusType.loading:
            return action.payload;
        default:
            return state;
    }
}

export default statusReducer;