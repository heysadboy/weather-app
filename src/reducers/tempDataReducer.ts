import { EActionType } from '../utils/enums';
import { IAction } from '../utils/interfaces';


const tempDataReducer = (state = [], action: IAction) => {
    switch (action.type) {
        case EActionType.temp_data:
            return action.payload;
        default:
            return state;
    }
};

export default tempDataReducer;