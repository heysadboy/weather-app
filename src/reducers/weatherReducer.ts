import { EActionType } from '../utils/enums';
import { IAction } from '../utils/interfaces';


const weatherReducer = (state = {}, action: IAction) => {
    switch (action.type) {
        case EActionType.fetch_weather:
            return action.payload;
        default:
            return state;
    }
};

export default weatherReducer;