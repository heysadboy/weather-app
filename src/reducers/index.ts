import { combineReducers } from 'redux';
import pageReducer from './pageReducer';
import tempDataReducer from './tempDataReducer';
import weatherReducer from './weatherReducer';

export default combineReducers({
    weather: weatherReducer,
    temparture: tempDataReducer,
    page: pageReducer
});