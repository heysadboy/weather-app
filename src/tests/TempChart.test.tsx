import renderer from 'react-test-renderer';
import TempChart from '../components/TempChart';
import { ETempType } from '../utils/enums';
import reducers from '../reducers';
import { Provider } from 'react-redux';
import { createStore } from 'redux';


test('Test Temp Chart', () => {
    const tempType: ETempType = ETempType.c;
    const currentDay: string = "test";
    const store = createStore(reducers);

    const component = renderer.create(
        <Provider store={store}>
            <TempChart tempType={tempType} currentDay={currentDay} />
        </Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});