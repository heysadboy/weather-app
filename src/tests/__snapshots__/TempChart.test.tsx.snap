// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Temp Chart 1`] = `
<div
  className="ui center aligned grid"
  id="temp-container"
>
  <div
    className="recharts-wrapper"
    style={
      Object {
        "cursor": "default",
        "height": 500,
        "position": "relative",
        "width": 800,
      }
    }
  >
    <svg
      className="recharts-surface"
      height={500}
      id="temp-chart"
      version="1.1"
      viewBox="0 0 800 500"
      width={800}
    >
      <defs>
        <clipPath
          id="temp-chart-clip"
        >
          <rect
            height={460}
            width={790}
            x={5}
            y={5}
          />
        </clipPath>
      </defs>
      <g
        className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
      >
        <line
          className="recharts-cartesian-axis-line"
          fill="none"
          height={30}
          orientation="bottom"
          stroke="#666"
          type="category"
          width={790}
          x={5}
          x1={5}
          x2={795}
          y={465}
          y1={465}
          y2={465}
        />
        <g
          className="recharts-cartesian-axis-ticks"
        >
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              orientation="bottom"
              stroke="#666"
              type="category"
              width={790}
              x={5}
              x1={202.5}
              x2={202.5}
              y={465}
              y1={471}
              y2={465}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              orientation="bottom"
              stroke="none"
              textAnchor="middle"
              type="category"
              width={790}
              x={202.5}
              y={473}
            >
              <tspan
                dy="0.71em"
                x={202.5}
              >
                0
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              orientation="bottom"
              stroke="#666"
              type="category"
              width={790}
              x={5}
              x1={597.5}
              x2={597.5}
              y={465}
              y1={471}
              y2={465}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              orientation="bottom"
              stroke="none"
              textAnchor="middle"
              type="category"
              width={790}
              x={597.5}
              y={473}
            >
              <tspan
                dy="0.71em"
                x={597.5}
              >
                auto
              </tspan>
            </text>
          </g>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
